HW07-ATM/src/main/java/ru/otus/hw07/ATM.java

```diff
@@ -0,0 +1,74 @@
+package ru.otus.hw07;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.Currency;
+import java.util.EnumSet;
+
+public class ATM {
+
+  private BinFactory binFactory;
+
+  public static BinFactory makeFactory(Currency currency) {
+    switch (currency.getCurrencyCode()) {
+      case "USD":
+        return new USDBinFactory(1000L);
+      default:
+        throw new IllegalArgumentException("KingdomType not supported.");
```

Comment:
```suggestion
        throw new IllegalArgumentException("Type not supported - " + currency.getCurrencyCode());
```

Правильно, что обрабатываете `default` и бросаете исключение, только "KingdomType" может запутать.